js/preload/preloadMain.js

```diff
@@ -213,6 +214,36 @@ function createAPI() {
 
     };
 
+    /**
+     * Open file in default app.
+     */
+    function openFile(id) {
```

Comment:
also i to make things a bit cleaner, can we move all of this code here related to download file into a separate module.  then require it from preload.js.